### PR TITLE
Removing FLux and introducing redux 2.0

### PR DESCRIPTION
**Pull request by jbb4play**
Flux has been removed.
Redux has been introduced.
Unit tests have been fixed to work with redux.
ProductSearch no longer has any component states, it is all handled by redux.
TwoDMap uses  the redux store to stay updated. 
Some nasty pieces of code have been refactured.

### CODE REVIEW

File: src/components/2dmap/TwoDimensionalMap.tsx [hunk 1]
```diff
@@ -61,35 +74,39 @@ export default class TwoDimensionalMap extends React.Component<ITwoDimensionalMa
    * Lifecycle react method. This method is triggered when the react component is correctly loaded into the dom.
    */
   public componentDidMount() {
-    this.generateMap();
+    const { selectedProduct, products } = this.props.productData;
+    this.generateMap(selectedProduct, products);
+  }
+
+  public componentWillReceiveProps(nextprops: props) {
```
Comment (sixfourtwo) [inline src/components/2dmap/TwoDimensionalMap.tsx]: I'd like to know why both componentDidMount() and componentWillRecieveProps() are needed when they do the same thing.

File: src/components/2dmap/TwoDimensionalMap.tsx [hunk 2]
```diff
@@ -261,44 +255,6 @@ export default class TwoDimensionalMap extends React.Component<ITwoDimensionalMa
...
     }
   }
-
-  /**
-   * Finds the dimensions of a polygon
-   * from 0,0 to the largest x and y
-   * @param polygon the outer polygon
-   * @returns A coordinate object, that contains the length and height of the polygon
-   */
-  public findDimensions(polygon: IPolygon): ICoord {
```
Comment (sixfourtwo) [inline src/components/2dmap/TwoDimensionalMap.tsx]: Is it on purpose to delete findDimensions()?

Comment (sixfourtwo) [reply src/components/2dmap/TwoDimensionalMap.tsx]: I see that you have put it in a helper class instead - please explain this choice. :)

File: src/components/2dmap/TwoDimensionalMap.tsx (hunk 1 above)
Comment (jbb4play) [reply src/components/2dmap/TwoDimensionalMap.tsx]: Without loading the map the first time aroound it wont load until you have performed a search

File: src/components/2dmap/TwoDimensionalMap.tsx (hunk 2 above)
Comment (jbb4play) [reply src/components/2dmap/TwoDimensionalMap.tsx]: Now that we itbis connected to the store usinh connect(... map state to props and map dispatch to props) you cant instaciate the class like you did before and also it is very bad practice yo instaciate a react component lile that therefore i moved it and tested it on its own

File: src/components/2dmap/TwoDimensionalMap.tsx (hunk 1 above)
Comment (jbb4play) [reply src/components/2dmap/TwoDimensionalMap.tsx]: The component renders when its load it, but at that point the data is not there yet, you must therefore tell it to render afrer you have received the data for the map

**Pull request merged**